eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -5,12 +5,22 @@ parameters:
     default: ''
 
 steps:
+  # We set DOTNET_ROLL_FORWARD so that .NET assemblies targeting older versions of .NET can run on newer ones.
```

Comment:
What prompted the need for this? We should try setting multilevellookup before doing this forced roll forward https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/use-dotnet-v2?view=azure-pipelines

Reply:
@weshaggard this:
- https://github.com/Azure/azure-sdk-tools/pull/5328#pullrequestreview-1290001058
- 
Basically @ckairen wants to target .NET 7. At first the idea was to use [`DotNetCoreVersion: 7.0.x`](https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/steps/install-dotnet.yml#L26) but it didn't work, as can be seen here:
- https://github.com/Azure/azure-sdk-tools/pull/5379#issuecomment-1423312514

So I proposed an alternative approach where we install .NET 7 instead of .NET 6 and make everything use it, even if it targets .NET 6.

But I understand you would prefer to keep `global.json` to `.NET 6.0.x`, but for @ckairen's tool use `DotNetCoreVersion: 7.0.x` and just install both .NETs ? I understand the `performMultiLevelLookup` would help in finding the right framework.